### PR TITLE
*: refactor GetColumnIndex() and some function

### PR DESCRIPTION
**Pull request by winoros**
In code, there is something like `expression.NewSchema(p.groupByCols).GetColumnIndex(colOriginal)`. But it's not necessary to create a Schema. So i change this method to `GetColumnIndex(cols []*Column, col *Column)`

### CODE REVIEW

**Comment by winoros** [comment]
PTAL @coocood @hanfei1991 @shenli @zimulala @tiancaiamao

File: plan/aggregation_push_down.go [hunk 1]
```diff
@@ -281,13 +281,13 @@ func (a *aggPushDownSolver) pushAggCrossUnion(agg *Aggregation, unionSchema expr
...
 		for _, arg := range newAggFunc.GetArgs() {
-			newArgs = append(newArgs, expression.ColumnSubstitute(arg, unionSchema, expression.Column2Exprs(unionChild.GetSchema().Columns)))
+			newArgs = append(newArgs, expression.ColumnSubstitute(arg, unionSchema.Columns, expression.Column2Exprs(unionChild.GetSchema().Columns)))
 		}
 		newAggFunc.SetArgs(newArgs)
 		newAgg.AggFuncs = append(newAgg.AggFuncs, newAggFunc)
 	}
 	for _, gbyExpr := range agg.GroupByItems {
-		newExpr := expression.ColumnSubstitute(gbyExpr, unionSchema, expression.Column2Exprs(unionChild.GetSchema().Columns))
+		newExpr := expression.ColumnSubstitute(gbyExpr, unionSchema.Columns, expression.Column2Exprs(unionChild.GetSchema().Columns))
```
Comment (hanfei1991) [inline plan/aggregation_push_down.go]: Maybe implement a schema 2 exprs is better?

File: expression/column.go [hunk 2]
```diff
@@ -55,15 +55,15 @@ func (col *CorrelatedColumn) IsCorrelated() bool {
 }
 
 // Decorrelate implements Expression interface.
-func (col *CorrelatedColumn) Decorrelate(schema Schema) Expression {
-	if schema.GetColumnIndex(&col.Column) == -1 {
+func (col *CorrelatedColumn) Decorrelate(cols []*Column) Expression {
```
Comment (hanfei1991) [inline expression/column.go]: I think it's ok for these function to receive Schema directly. Only GetColumnIndex is needed.

File: plan/aggregation_push_down.go (hunk 1 above)
Comment (winoros) [inline plan/aggregation_push_down.go]: After adding unique key info into schema, columns2expr will be more accurate than schema2expr.

**Comment by hanfei1991** [comment]
This change may make the code more complex. So close it.

**Pull request closed without merging**